components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/GraphQLQueryAnalysisHandler.java

```diff
@@ -54,7 +57,13 @@ public boolean handleRequest(MessageContext messageContext) {
             queryAnalyzer = new QueryAnalyzer(schema);
         }
         String payload = messageContext.getProperty(APIConstants.GRAPHQL_PAYLOAD).toString();
-        if (!isDepthAndComplexityValid(messageContext, payload)) {
+        Map<String, Object> variableMap;
+        if(messageContext.getPropertyKeySet().contains("VARIABLE_MAP")){
```

Comment:
Define this as constant

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/226c49a261240dc5305f96d69aadb6f86edfae84